src/packages/documents/documents/components/input-document-picker-root/input-document-picker-root.element.ts

```diff
@@ -1,101 +1,268 @@
-import { UmbDocumentPickerContext } from '../input-document/input-document.context.js';
-import type { UmbDocumentItemModel } from '../../repository/index.js';
-import { html, customElement, property, state, ifDefined, repeat } from '@umbraco-cms/backoffice/external/lit';
+import { html, css, customElement, property, ifDefined, state, repeat } from '@umbraco-cms/backoffice/external/lit';
 import { FormControlMixin } from '@umbraco-cms/backoffice/external/uui';
+import { UmbChangeEvent } from '@umbraco-cms/backoffice/event';
+import { UmbId } from '@umbraco-cms/backoffice/id';
 import { UmbLitElement } from '@umbraco-cms/internal/lit-element';
-import type { DocumentItemResponseModel } from '@umbraco-cms/backoffice/backend-api';
+import { UmbSorterController } from '@umbraco-cms/backoffice/sorter';
+import { umbExtensionsRegistry } from '@umbraco-cms/backoffice/extension-registry';
+import { UMB_MODAL_MANAGER_CONTEXT } from '@umbraco-cms/backoffice/modal';
+import {
+	UMB_DYNAMIC_ROOT_ORIGIN_PICKER_MODAL,
+	UMB_DYNAMIC_ROOT_QUERY_STEP_PICKER_MODAL,
+} from '@umbraco-cms/backoffice/dynamic-root';
+import type {
+	ManifestDynamicRootOrigin,
+	ManifestDynamicRootQueryStep,
+} from '@umbraco-cms/backoffice/extension-registry';
+import type { UmbModalContext, UmbModalManagerContext } from '@umbraco-cms/backoffice/modal';
+import type { UmbTreePickerDynamicRoot, UmbTreePickerDynamicRootQueryStep } from '@umbraco-cms/backoffice/components';
+import { query } from '@umbraco-cms/backoffice/router';
 
 @customElement('umb-input-document-picker-root')
 export class UmbInputDocumentPickerRootElement extends FormControlMixin(UmbLitElement) {
-	public get unique(): string | null | undefined {
-		return this.#documentPickerContext.getSelection()[0];
-	}
-	public set unique(unique: string | null | undefined) {
-		const selection = unique ? [unique] : [];
-		this.#documentPickerContext.setSelection(selection);
+	protected getFormElement() {
+		return undefined;
 	}
 
-	@property()
-	public set value(unique: string) {
-		this.unique = unique;
-	}
+	@state()
+	private _originManifests: Array<ManifestDynamicRootOrigin> = [];
 
 	@state()
-	private _items?: Array<UmbDocumentItemModel>;
+	private _queryStepManifests: Array<ManifestDynamicRootQueryStep> = [];
+
+	@property({ attribute: false })
+	data?: UmbTreePickerDynamicRoot | undefined;
 
-	#documentPickerContext = new UmbDocumentPickerContext(this);
+	#dynamicRootOrigin?: { label: string; icon: string; description?: string };
 
-	// TODO: DynamicRoot - once feature implemented, wire up context and picker UI. [LK]
-	#dynamicRootPickerContext = {
-		openPicker: () => {
-			throw new Error('DynamicRoot picker has not been implemented yet.');
-		},
-	};
+	#modalContext?: UmbModalManagerContext;
```

Comment:
Consider inferring the type from the token:

```suggestion
	#modalContext?: typeof UMB_MODAL_MANAGER_CONTEXT.TYPE;
```

Reply:
Done. 👍